### ISSUE TITLE
Incorrect column name in presence of skipped columns

### ISSUE THREAD

**Issue by jimhester** [r-lib/vroom#293]
It seems the column names are applied before skipping, rather than afterwards. e.g.

``` r
res <- vroom::vroom("1,2,3\n4,5,6", col_names = "z", col_types = "__i")

testthat::expect_equal(names(res), "z")
#> Error: names(res) not equal to "z".
#> 1/1 mismatches
#> x[1]: "X3"
#> y[1]: "z"
```

<sup>Created on 2021-01-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jimhester** [comment]
This is not very simple to resolve, we currently expand the names to the size of the full columns, then standardise the column types. So there is a chicken and the egg problem where we need the expanded names to standardize the columns, but need the un-expanded names to assign them to skipped columns appropriately.

**Comment by jimhester** [comment]
Fixed by 9ab95eaf8297cf4bd550982b64411c8139ac8915
